lodepng_fuzzer.cpp

```diff
@@ -59,8 +59,6 @@ unsigned testDecode(lodepng::State& state, const uint8_t* data, size_t size) {
 extern "C" int LLVMFuzzerTestOneInput(const uint8_t* data, size_t size) {
   if(size == 0) return 0;
   size_t random_color_type = data[0] % num_combinations;
-  data++;
```

Comment:
We probably want to keep this as-is. The fuzzer was essentially consuming the first bit of random data for the color type. If we have valid PNG files, this will always have the same value if it isn't consumed.

Reply:
Hi, I tried to revert the change back only for this line: 
`data++`
Then again the fuzzer is not able to cover the valid png code. I don't think we should make any change to data variable so that at least some files are considered as valid pngs. However, we can choose to get the color by some other means.

Reply:
You could compute the variable in a different way but it's important that you not simply take data[0] for valid png inputs. In almost all cases we'd just use the value of the first byte of the PNG header in that case. (You could use a hash, or modify the corpus to include an extra first byte in each file).

Reply:
Modifying the corpus would not be difficult to manage further if we would like to change the seed corpus. I tried to implement hashing and the coverage improved but by changing the `random_color_type` to the last byte of the input, the results are somewhat better as them libFuzzer have more control what kind of data to parse.